lbuild/api.py

```diff
@@ -77,6 +77,14 @@ def __init__(self, cwd=None, config=None, options=None, collectors=None):
         self.config.add_commandline_collectors(listify(collectors))
         self.parser = Parser(self.config)
 
+        self.outpath = outpath
+        # set special lbuild options from other sources
+        # special xml options:
+        special_outpath = self.config.options.get("lbuild:path", (None, None))[0]
+        if special_outpath:
+            self.outpath = special_outpath.strip()
+            del self.config.options["lbuild:path"]
+
```

Comment:
I was thinking something like a top-level `<outpath>path</outpath>` analog to `<extends>config</extends>`.
See https://github.com/modm-io/lbuild/blob/develop/lbuild/config.py#L187.

Reply:
There should also be a unit test for this around here: https://github.com/modm-io/lbuild/blob/develop/test/config_test.py#L107-L162

Reply:
> I was thinking something like a top-level <outpath>path</outpath> analog to <extends>config</extends>.

At first I implemented something special, but then I decided to go with the default configuration theme you have for every other module. I.e. defining an option an deciding whom this option belongs to via namespaces ... So <option name="lbuild:path> myPath </option> was born

Reply:
No, it cannot be an option, it is completely inconsistent with the rest of lbuild and it breaks the API: `lbuild --path path1/ -D lbuild:path=path2/ build`. It needs to be it's separate thing and not comandeer the options for that.

Reply:
Okay that makes sense l, I'm gonna put into a
`<outpath>path</outpath>`
And see where this is going